src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -94,50 +96,30 @@ private void Start(CancellationToken cancellationToken)
                     return;
                 }
 
-                OpeningPoint = snapshot.CreateTrackingPoint(openingSnapshotPoint, PointTrackingMode.Positive);
+                OpeningPoint = SubjectBuffer.CurrentSnapshot.CreateTrackingPoint(openingSnapshotPoint, PointTrackingMode.Positive);
```

Comment:
since we are accessing .CurrentSnapshot, it would be a good idea for this type to either derive from, or contain a ForegroundThreadAffinitizedObject.  We woudl then assert in these methods htat were were called on the UI thread.  thus expressing hte necessary constraints for our caller, and throwing immediately if that is ever violated.